### PR TITLE
New Domain classes School and Vehicle

### PR DESCRIPTION
**Pull request by BV135**
What I did:
- Add VehicleServiceImpl with full CRUD operations
- Add SchoolServiceImpl with full CRUD operations  
- Create VehicleServiceImplTest with 14 comprehensive tests
- Create SchoolServiceImplTest with 16 comprehensive tests
- Fix type conversion issues in UserServiceImpl
- Implement proper Mockito-based unit testing
- Extend base Service interface for clean architecture
- Add proper error handling and validation
- Test coverage for edge cases and error scenarios"

### CODE REVIEW

**Review by Casey-Nolte — APPROVED** [review]

File: src/main/java/com/debuggers/DemoApplication.java [hunk 1]
```diff
@@ -1,13 +1,68 @@
...
 
 	public static void main(String[] args) {
 		SpringApplication.run(DemoApplication.class, args);
 	}
 
-}
+	@Override
+	public void run(String... args) {
+
+		// Create a user [Works!]
```
Comment (EthanLeRoux) [inline src/main/java/com/debuggers/DemoApplication.java]: Why code in Main?

File: src/main/java/com/debuggers/DemoApplication.java [hunk 2]
```diff
@@ -1,13 +1,68 @@
...
+		try {
+			if (Desktop.isDesktopSupported()) {
+				Desktop.getDesktop().browse(new URI("http://localhost:8081"));
+			} else {
+				System.out.println("Desktop not supported.");
+			}
+		} catch (Exception e) {
+			System.err.println("Failed to open browser: " + e.getMessage());
+		}
+	}
```
Comment (EthanLeRoux) [inline src/main/java/com/debuggers/DemoApplication.java]: Was this for testing?

File: src/main/java/com/debuggers/controller/UserController.java
Comment (EthanLeRoux) [inline src/main/java/com/debuggers/controller/UserController.java:1]: Is this a redo of the class?
User controller was created by @BV135 before already

File: src/main/java/com/debuggers/domain/User.java
Comment (EthanLeRoux) [inline src/main/java/com/debuggers/domain/User.java:1]: is this a rework?

File: src/main/java/com/debuggers/repository/UserRepository.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+    */
+import com.debuggers.domain.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.util.Set;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, String> {
```
Comment (EthanLeRoux) [inline src/main/java/com/debuggers/repository/UserRepository.java]: User Id is spose to be Long

File: src/main/resources/application.properties
Comment (EthanLeRoux) [inline src/main/resources/application.properties:1]: application properties shouldnt be pushed unless its global changes we need in the app, can add to ur gitignore

**Review by EthanLeRoux — CHANGES_REQUESTED** [review]
Tests arent necessary while we develop for next week monday. its gonna slow us down for now, try testing with postman.
